### PR TITLE
test: Use test framework utils in functional tests

### LINKED ISSUE #23119: test: replace bare asserts with assertion helpers (assert_equal() etc.)
#23117 replaced asserts with the test framework's internal helpers (see https://github.com/bitcoin/bitcoin/blob/dccf3d25f9e78909eb7b3143e89a7c87fac25ab5/test/functional/test_framework/util.py#L47-L59) for a single test, in order to see the expected and failed values if such an assertion fails. The same should be done for all the remaining tests. Potential candidates can be found via
```
$ cd ./test/functional
$ git grep assert.*==
```

#### Useful skills:

basic Python3 knowledge

#### Want to work on this issue?

For guidance on contributing, please read [CONTRIBUTING.md](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md) before opening your pull request.

### PR DESCRIPTION
**Pull request by osagie98**
Replaces bare asserts with test framework utils across both the functional tests and the test framework itself.

Also adds the `assert_not_equal`, `assert_less_than`, and `assert_less_than_or_equal` util functions for greater readability.

Fixes #23119.

### CODE REVIEW

**Comment by osagie98** [comment]
Though the change is simple, it touches a ton of files. I can break it into multiple PRs if that makes sense.

**Comment by theStack** [comment]
@osagie98: Can you squash the commits and rebase on master? Happy to review.

**Comment by osagie98** [comment]
@theStack done (I hope, still getting used to the GitHub workflow). Let me know if you'll need anything else

**Comment by osagie98** [comment]
Hi @theStack, any chance you'd be able to review this?

**Comment by theStack** [comment]
> Hi @theStack, any chance you'd be able to review this?

Yes, planning to (finally) review this within the next 1-2 days.

File: test/functional/feature_coinstatsindex.py [hunk 1]
```diff
@@ -226,7 +228,7 @@ def _test_coin_stats_index(self):
 
         self.generate(index_node, 1, sync_fun=self.no_op)
         res10 = index_node.gettxoutsetinfo('muhash')
-        assert res8['txouts'] < res10['txouts']
+        assert_greater_than(res10['txouts'], res8['txouts'])
```
Comment (theStack) [inline test/functional/feature_coinstatsindex.py:231]: I noticed that here (and at some other places), `assert x < y` is replaced by `assert_greater_than(y, x)` rather than the more obvious `assert_less_than(x, y)`. Is that change intentional or is that a left-over from an earlier version, when `assert_left_than` was not introduced yet?

Comment (osagie98) [reply test/functional/feature_coinstatsindex.py:231]: Yes, early on I kept only `assert_greater_than()`, before I decided that `assert_less_than()`was needed. I can go back and fix those to make it clear.

Comment (osagie98) [reply test/functional/feature_coinstatsindex.py]: Done. Force pushed to allow for the rebase.

**Review by BrandonOdiwuor — COMMENTED** [review]
Concept ACK

This PR improves functional tests (readability) by using/introducing complimentary util functions to `assert` ie `assert_equal`, `assert_not_equal`, `assert_less_than`,    `assert_less_than_or_equal`

**Review by BrandonOdiwuor — COMMENTED** [review]
There are still some instances of bare assert which could be replaced by the helpers such as in `rpc_psbt.py`, `wallet_disable.py` is this intentional?

**Comment by osagie98** [comment]
Hi @BrandonOdiwuor,

Sorry, I missed the notification of your messages. I missed a few asserts, so I went back to add them. I also initially chose not to change any `assert x is True/False` as I wasn't sure if it would be useful, but I decided now to change those as well.

Please take another look whenever you're free, thank you!

File: test/functional/wallet_miniscript.py [hunk 2]
```diff
@@ -275,7 +276,8 @@ def signing_test(
         self.wait_until(lambda: txid in self.funder.getrawmempool())
         self.funder.generatetoaddress(1, self.funder.getnewaddress())
         utxo = self.ms_sig_wallet.listunspent(addresses=[addr])[0]
-        assert txid == utxo["txid"] and utxo["solvable"]
+        assert_equal(txid == utxo["txid"])
```
Comment (kevkevinpal) [inline test/functional/wallet_miniscript.py:279]:
```suggestion
        assert_equal(txid, utxo["txid"])
```

File: test/functional/test_framework/util.py [hunk 3]
```diff
@@ -67,6 +72,16 @@ def assert_greater_than_or_equal(thing1, thing2):
         raise AssertionError("%s < %s" % (str(thing1), str(thing2)))
 
 
+def assert_less_than(thing1, thing2):
+    if thing1 >= thing2:
+        raise AssertionError("%s .= %s" % (str(thing1), str(thing2)))
```
Comment (stratospher) [inline test/functional/test_framework/util.py:77]: 2103e12: here and for `assert_less_than_or_equal` too.
```suggestion
        raise AssertionError("%s >= %s" % (str(thing1), str(thing2)))
```

File: test/functional/test_framework/util.py [hunk 4]
```diff
@@ -57,6 +57,11 @@ def assert_equal(thing1, thing2, *args):
         raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
 
 
+def assert_not_equal(thing1, thing2, *args):
```
Comment (stratospher) [inline test/functional/test_framework/util.py:60]: 2103e12: maybe just have `assert_not_equal(thing1, thing2)`. i couldn't find an occurrence where we needed to `assert_not_equal` 3+ things. ex: `assert_not_equal(2, 2, 3)`

Comment (stratospher) [reply test/functional/test_framework/util.py:60]: i think you can find more places where `assert_not_equal` can be used using the command @theStack mentioned [here](https://github.com/bitcoin/bitcoin/issues/23119#issue-1009729239)!

example first 3 lines output:
```
$ git grep "assert ".*\!=
feature_fee_estimation.py:        assert fee_dat_current_content != fee_dat_initial_content
feature_fee_estimation.py:        assert fee_dat_current_content != fee_dat_initial_content
mempool_accept_wtxid.py:        assert child_one_wtxid != child_two_wtxid
```
